### PR TITLE
[New-IR] add pass registry

### PR DESCRIPTION
**Pull request by zhiqiu**
### PR types

New features
### PR changes

Others
### Description

[New-IR] add pass registry
Pcard-67164

### CODE REVIEW

File: paddle/fluid/pybind/ir.cc [hunk 1]
```diff
@@ -514,7 +508,8 @@ void BindPassManager(pybind11::module *m) {
           py::arg("opt_level") = 2)
       .def("add_pass",
            [](PassManager &self, std::string pass_name) {
```
Comment (yuanlehome) [inline paddle/fluid/pybind/ir.cc:510]: std::string作为函数参数，最好还是const&吧

File: test/ir/new_ir/test_pass_manager.py [hunk 2]
```diff
@@ -51,11 +51,12 @@ def test_op(self):
         self.assertTrue('pd.uniform' in op_names)
         pm = ir.PassManager()
         pm.add_pass(
-            'DeadCodeEliminationPass'
+            'dead_code_elimination'
         )  # apply pass to elimitate dead code
         pm.run(new_program)
         op_names = [op.name() for op in new_program.block().ops]
         # print(op_names)
+        # TODO(zhiqiu): unify the name of pass
```
Comment (yuanlehome) [inline test/ir/new_ir/test_pass_manager.py:59]: 可以考虑所有的pass name都统一改为小写加下划线形式

**Review by yuanlehome — APPROVED** [review]
LGTM

File: test/ir/new_ir/test_pass_manager.py (hunk 2 above)
Comment (zhiqiu) [reply test/ir/new_ir/test_pass_manager.py:59]: fix it in the next pr

File: paddle/fluid/pybind/ir.cc (hunk 1 above)
Comment (zhiqiu) [reply paddle/fluid/pybind/ir.cc:510]: fix it in the next pr

**Pull request merged**